components/src/dynamo/sglang/request_handlers/handler_base.py

```
@@ -167,43 +194,54 @@ async def release_memory_occupation(self, body: dict) -> dict:
-            # Step 1: Unregister endpoint from discovery FIRST
             try:
-                await self.generate_endpoint.unregister_endpoint_instance()
-            except Exception as unreg_err:
-                logging.warning(
-                    f"Failed to unregister endpoint from discovery: {unreg_err}"
+                # Stop new requests and drain in-flight work once when entering released state.
+                if not self._released_memory_tags:
+                    if self.generate_endpoint is not None:
+                        try:
+                            await self.generate_endpoint.unregister_endpoint_instance()
+                        except Exception as unreg_err:
```

Ruff (0.15.2)

[warning] Do not catch blind exception: `Exception`

(BLE001)